vlm/src/test/scala/geotrellis/contrib/vlm/gdal/GDALWarpOptionsSpec.scala

```diff
@@ -52,46 +55,40 @@ class GDALWarpOptionsSpec extends FunSpec with RasterMatchers with BetterRasterM
       targetCRS = None
     )
 
-  def dsreproject(dataset: Dataset): Dataset =
-    GDAL.warp("", dataset, reprojectOptions, None)
-  def dsresample(dataset: Dataset, uri: Option[String]): Dataset =
-    GDAL.warp("", dataset, resampleOptions, uri.map(str => str -> List(reprojectOptions)))
+  def rasterSourceFromUriOptions(uri: String, options: GDALWarpOptions) = {
+    GDALRasterSource(uri, options)
+  }
 
-  def dsreprojectOpt(dataset: Dataset): Dataset = {
+  def dsreprojectOpt(uri: String) = {
     val opts =
-      GDALWarpOptions(Some("MEM"))
+      GDALWarpOptions()
         .reproject(
           rasterExtent = GridExtent(Extent(630000.0, 215000.0, 645000.0, 228500.0), 10, 10).toRasterExtent,
           CRS.fromString("+proj=lcc +lat_1=36.16666666666666 +lat_2=34.33333333333334 +lat_0=33.75 +lon_0=-79 +x_0=609601.22 +y_0=0 +datum=NAD83 +units=m +no_defs "),
           WebMercator,
           ReprojectOptions.DEFAULT.copy(targetCellSize = CellSize(10, 10).some)
         )
-
-    GDAL.warp("", dataset, opts, None)
+    rasterSourceFromUriOptions(uri, opts)
   }
-  def dsresampleOpt(dataset: Dataset): Dataset = {
+
+  def dsresampleOpt(uri: String) = {
     val opts =
-      GDALWarpOptions(Some("MEM"))
+      GDALWarpOptions()
         .reproject(
           rasterExtent = GridExtent(Extent(630000.0, 215000.0, 645000.0, 228500.0), 10, 10).toRasterExtent,
           CRS.fromString("+proj=lcc +lat_1=36.16666666666666 +lat_2=34.33333333333334 +lat_0=33.75 +lon_0=-79 +x_0=609601.22 +y_0=0 +datum=NAD83 +units=m +no_defs "),
           WebMercator,
           ReprojectOptions.DEFAULT.copy(targetCellSize = CellSize(10, 10).some)
         )
-        .resample(
```

Comment:
how do you test `reproject.resample` thing? Otherwise how the tests pass?

Reply:
This is one of the questions that I had.  I was getting a different region from merging the warp options and passing them to GDAL than this code was producing.  Removing it caused them to match -- I want to talk about that.

Reply:
Okay, thanks for the talk, I understand what is happening here, now.

Reply:
Okay, addressed.